klog.go

```diff
@@ -1165,9 +1171,18 @@ func (l *loggingT) createFiles(sev severity) error {
 const flushInterval = 5 * time.Second
 
 // flushDaemon periodically flushes the log file buffers.
-func (l *loggingT) flushDaemon() {
-	for range time.NewTicker(flushInterval).C {
-		l.lockAndFlushAll()
+// added select case to quit channels
+func (l *loggingT) flushDaemon(stop, done chan struct{}) {
+	ticker := time.NewTicker(flushInterval)
+	for {
+		select {
+		case <-stop:
+			continue
+		case <-ticker.C:
+			l.lockAndFlushAll()
+			close(done)
+			return
```

Comment:
Previously this was flushing the logs every ticker interval, but now it appears to only flush the logs on the first interval and return early. This seems like it would lead to the logs not being flushed after the initial interval passes.

Reply:
I will look for return modify the same accordingly. But previous code have a leak on go routine. The init() function will exit before executing https://github.com/kubernetes/klog/blob/f81e6a7b1a3fc2001c75aedc49c82442ceaa3c5e/klog.go#L417 go routine on logs in current code snippet. The value sent on ticker channel after interval will never get received since the init function will exit. I am not sure the logs will even execute more than once, even if the go routine will complete execution once which might not be the case.

Reply:
@detiber Thanks for informing about return. I will use a halt signal to exit else it will wait for the ticker channel for given interval and flush the logs. Please let me know in which case you want to stop the go routine. If it is an ongoing process, there is no need and it will wait for ticker channel to have a value after and interval when it does, the logs will be flushed.